src/group_theory/order_of_element.lean

```diff
@@ -181,6 +181,27 @@ end
 @[simp] lemma order_of_eq_one_iff : order_of a = 1 ↔ a = 1 :=
 ⟨λ h, by conv_lhs { rw [← pow_one a, ← h, pow_order_of_eq_one] }, λ h, by simp [h]⟩
 
+@[simp] lemma order_of_submonoid {G : Type*} [monoid G] {H : submonoid G} (σ : H) :
+  order_of (σ : G) = order_of σ :=
```

Comment:
A good lemma: `order_of x = order_of y ↔ (∀ n, x ^ n = 1 ↔ y ^ n = 1)`

Reply:
Thanks. Using that lemma actually shortened the whole PR by 7 lines!